app/packages/looker/src/overlays/base.ts

```diff
@@ -93,20 +93,50 @@ export abstract class CoordinateOverlay<
     return state.options.selectedLabels.includes(this.label.id);
   }
 
-  getColor({ options: { coloring } }: Readonly<State>): string {
+  getColor({
+    options: { coloring, customizeColorSetting },
+  }: Readonly<State>): string {
     let key;
+    let pool;
 
     switch (coloring.by) {
       case "field":
+        // check if the field has a customized color, use it if it is a valid color
+        const fieldColor = customizeColorSetting.find(
+          (s) => s.field === this.field
+        )?.fieldColor;
```

Comment:
nit: I am unfamiliar with this code. hopefully the computation here is small enough not to impact rendering performance